File: scrimage-core/src/main/scala/com/sksamuel/scrimage/canvas/drawable/FilledOval.scala
package com.sksamuel.scrimage.canvas.drawable

import java.awt.Graphics2D

import com.sksamuel.scrimage.canvas.{Drawable, GraphicsContext}

case class FilledOval(x: Int, y: Int, width: Int, height: Int, context: GraphicsContext) extends Drawable {
  def draw(g: Graphics2D): Unit = g.fillOval(x, y, width, height)
  def outline: Oval = Oval(x, y, width, height, context)
}

File: scrimage-core/src/main/java/com/sksamuel/scrimage/canvas/drawables/FilledOval.java
package com.sksamuel.scrimage.canvas.drawables;

import com.sksamuel.scrimage.canvas.Drawable;
import com.sksamuel.scrimage.canvas.GraphicsContext;
import com.sksamuel.scrimage.canvas.RichGraphics2D;

public class FilledOval implements Drawable {

    private final int x;
    private final int y;
    private final int width;
    private final int height;
    private final GraphicsContext context;

    public FilledOval(int x, int y, int width, int height, GraphicsContext context) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
        this.context = context;
    }

    @Override
    public void draw(RichGraphics2D g) {
        g.fillOval(x, y, width, height);
    }

    public Oval toOutline() {
        return new Oval(x, y, width, height, context);
    }

    @Override
    public GraphicsContext context() {
        return context;
    }
}
